flipperdevices/flipperzero-firmware
# [FL-3918] Full-fledged JS SDK + npm packages

# What's new
  - Added npm packages: `@flipperdevices/fz-sdk` (contains the SDK, i.e. API typings and helper tooling), `@flipperdevices/create-fz-app` (interactive JS project creation wizard). These packages will be published manually by me if/when this pull request is merged.
  - JS API versioning system. Thank you to @Willy-JL for collaborating with us on this one.
  - Small naming/doc fixes: #3960, #3965
  - `var` and `const` keyword support in the interpreter
  - `Object.defineProperty` stub

# Verification 
  - Run unit tests
  - Install Node.js together with npm
  - Run the interactive project creation wizard:
    - If the packages do end up getting published, `npx @flipperdevices/create-fz-app@latest`
    - If not, `cd applications/system/js_app/packages && npm exec ./create-fz-app`. This will fail at the "Installing packages" step, which is expected if the package has not yet been published. Then, `cd my-flip-app` and change the version specifier in `package.json` for the `fz-sdk` package from `^0.1` to `link:../fz-sdk`
  - cd into the newly created app
  - Do `npm start`
  - Verify that the app launches on the Flipper

# Checklist (For Reviewer)
